test/HydraDelegation/HydraDelegation.test.ts

```diff
@@ -499,6 +589,93 @@ export function RunHydraDelegationTests(): void {
+        );
+
+        // claim & check balance
+        await expect(
+          vestManager.claimVestedPositionReward(delegatedValidator.address, epochNum, balanceChangeIndex)
+        ).to.not.emit(hydraDelegation, "CommissionClaimed");
+      });
+
+      it("should distribute the right commission to the validator on claim for vested position", async function () {
+        const { systemHydraChain, hydraStaking, hydraDelegation, vestManager } = await loadFixture(
+          this.fixtures.vestManagerFixture
+        );
+        const validator = this.signers.validators[2];
+        const oldCommission = 10;
+        await hydraDelegation.connect(validator).setCommission(oldCommission);
+        time.increase(DAY * 30);
+
+        await vestManager.openVestedDelegatePosition(validator.address, 1, {
+          value: this.minDelegation.mul(2),
+        });
+        // Commit epochs so rewards to be distributed
+        await commitEpochs(
+          systemHydraChain,
+          hydraStaking,
+          [this.signers.validators[0], this.signers.validators[1], validator],
+          5, // number of epochs to commit
+          this.epochSize
+        );
+
+        // make sure that the vested position does not take the current commission, but the one at the time of delegation
+        const newCommission = oldCommission * 3;
+        await hydraDelegation.connect(validator).setCommission(newCommission);
+
+        // commit epoch
+        await commitEpoch(
+          systemHydraChain,
+          hydraStaking,
+          [this.signers.validators[0], this.signers.validators[1], validator],
+          this.epochSize,
+          WEEK * 2 + DAY
+        );
+
+        // prepare params for call
+        const { epochNum, balanceChangeIndex } = await getClaimableRewardRPSData(
+          systemHydraChain,
+          hydraDelegation,
+          validator.address,
+          vestManager.address
+        );
+
+        // check claimable rewards for the position
+        const expectedReward = await hydraDelegation.getRawReward(validator.address, vestManager.address);
+        const position = await hydraDelegation.vestedDelegationPositions(validator.address, vestManager.address);
+        const rewardApr = applyVestingAPR(position.base, position.vestBonus, position.rsiBonus, expectedReward);
+        const commission = calculateCommissionCutFromFinalReward(rewardApr, hre.ethers.BigNumber.from(oldCommission));
+
+        // claim & check balance
+        const balanceBefore = await validator.getBalance();
+        await expect(vestManager.claimVestedPositionReward(validator.address, epochNum, balanceChangeIndex)).to.emit(
+          hydraDelegation,
+          "CommissionClaimed"
+        );
+        const balanceAfter = await validator.getBalance();
+        expect(balanceAfter).to.be.gt(balanceBefore.add(commission).mul(99).div(100));
+        expect(balanceAfter).to.be.lt(balanceBefore.add(commission).mul(101).div(100));
+      });
     });
 
```

Comment:
adding tests for the view functions that handle commission now will be beneficial as well